providers/base/api/handlers/api_data_handler.go

```diff
@@ -23,3 +23,17 @@ type APIDataHandler[K providertypes.ResponseKey, V providertypes.ResponseValue]
 	// reflect either the time the data was fetched or the time the API last updated the data.
 	ParseResponse(ids []K, response *http.Response) providertypes.GetResponse[K, V]
 }
+
+// APIDataHandlerWithBody defines an interface that is implemented by some API providers
+// that depend on information transfer via HTTP request bodies. This is a wrapper over
+// the base APIDataHandler interface, with an additional method to fetch a request body
+//
+//go:generate mockery --name APIDataHandlerWithBody --output ./mocks/ --case underscore
+type APIDataHandlerWithBody[K providertypes.ResponseKey, V providertypes.ResponseValue] interface {
+	APIDataHandler[K, V]
+
+	// CreateBody is used to create an HTTP response body that will be embdedded in the HTTP
+	// request to the server defined by the CreateURL response. The HTTP body generated will
+	// depend on the ids (identifiers of data to fetch) passed to the implementation.
+	CreateBody(ids []K) ([]byte, error)
+}
```

Comment:
Why create an additional interface instead of just adding it to the other one? you already inherit `APIDataHandler` here. It should be relatively easy to update the existing API providers to just have no-ops for the `CreateBody` function.

Reply:
I do like the additional method tho.

Reply:
The outcomes are equivalent, this j ends up w/ less wasted code.

Reply:
Doing things the other way, I'd have to change all the existing API providers w/ a method that is inherently unused, seems weird to me